nucypher/network/nodes.py

```diff
@@ -827,11 +835,19 @@ def learn_from_teacher_node(self, eager=False, canceller=None):
             return RELAX
 
         try:
-            response = self.network_middleware.get_nodes_via_rest(
-                node=current_teacher,
-                announce_nodes=announce_nodes,
-                fleet_state_checksum=self.known_nodes.checksum,
+            optional_latency_context_manager = (
+                self.node_latency_collector.get_latency_tracker(
+                    current_teacher.checksum_address
+                )
+                if self.node_latency_collector
+                else contextlib.nullcontext()
             )
+            with optional_latency_context_manager:
+                response = self.network_middleware.get_nodes_via_rest(
+                    node=current_teacher,
+                    announce_nodes=announce_nodes,
+                    fleet_state_checksum=self.known_nodes.checksum,
```

Comment:
This is a reasonable place to gather latency samples - that being said are there any other options  we can consider?  Side note: based on the construction of the collector below it's important to only measure one request on a regular basis or else we'll get skewed results.  Am I understanding this correctly?

Reply:
Yep, you got it. In general, we need to be consistent when measuring "node latency", i.e. use the same measurement for all nodes.